core/src/main/java/org/mule/runtime/core/api/extension/ExtensionManager.java

```diff
@@ -124,6 +124,25 @@
   Optional<ConfigurationProvider> getConfigurationProvider(String configurationProviderName);
 
   /**
+   * Returns (if any) a {@link ConfigurationModel} chosen from the {@code assignableConfigurationModels} that can be used for
+   * obtaining a {@link ConfigurationProvider}. From the {@code assignableConfigurationModels}, if there is only one
+   * {@link ConfigurationProvider} registered that can be used, that will be returned. If there are more than one, an
+   * {@link IllegalStateException} will be thrown.
+   *
+   * If there are no {@link ConfigurationProvider} registered, if there is only {@link ConfigurationModel} that can be used
+   * implicitly, that will be returned. If there are more than one, an {@link IllegalStateException} will be thrown.
+   *
+   * If none of the previous conditions could be fulfilled, {@link Optional#empty()} will be returned.
+   * 
+   * @param extensionModel the {@link ExtensionModel} for which a configuration is wanted
+   * @param assignableConfigurationModels a {@link Set} of all the possible {@link ConfigurationModel} that can be used for a
+   *        given component.
+   * @return an {@link Optional} {@link ConfigurationModel}
+   */
+  Optional<ConfigurationModel> getConfigurationModelForExtension(ExtensionModel extensionModel,
```

Comment:
I don't thin this method should be exposed here. It makes sense for the ExtnesionManager to know how to do this, but it shouldn't expose that behaviour. This is very internal and this interface is API